### ISSUE TITLE
`Cannot resolve version: esprima-fb@undefined` in otherwise apparently working project

### ISSUE THREAD

**Issue by Ralith** [svanderburg/node2nix#23]
node2nix does not seem to be able to handle https://github.com/matrix-org/matrix-appservice-irc . `npm install` on the project works fine, fetching two different concrete versions of `esprima-fb` over the course of filling in all dependencies, but node2nix fails with `Cannot resolve version: esprima-fb@undefined`

**Comment by svanderburg** [comment]
Hmm super weird.

It appears that the corresponding package indirectly refers to the following version specifier: `~3001.0001.0000-dev-harmony-fb`. This version specifier appears to correspond to a semver version range.

However, the semver parser does not seem to accept it. The leading 0s seem to be a problem of the minor and patchlevel components.

I need to figure out how NPM handles this. node2nix uses the semver API to parse this version specifier, but that does not seem to work.

**Comment by svanderburg** [comment]
It seems that the loose interpretation parameter did the trick in rev 8acaa08fd412738e326a4a1cbfcbfcb1d75ba2be. It does not seem to be documented in the semver API (https://npmjs.com/package/semver), but after inspecting the code of the semver command-line utility, I figured out there's another optional parameter.

Now the esprima-fb dependency resolves properly.

Unfortunately, I ran into another issue with git. I need to work on a better version substitution mechanism to resolve this.
